core/metacling/src/TClingTypeInfo.cxx

```diff
@@ -256,7 +257,9 @@ std::string TClingTypeInfo::NormalizedName(const ROOT::TMetaUtils::TNormalizedCt
    if (!IsValid()) {
       return "";
    }
-   std::string buf;
+   TTHREAD_TLS_DECL(std::string, buf);
```

Comment:
Why does this need to be TLS, esp given that there's a lock? I'd also have thought that `GetNormalizedName` locks itself...?

Reply:
Maybe we want to place the locks in `TMetaUtils` functions rather than here?

Reply:
Another question for @pcanal ...

Reply:
The thread-local string buffer is only needed in functions that return a `const char *`, i.e. the pointer to the buffer. This one in particular returns a `std::string`, so no need for `thread_local`. The locks are indeed needed anywhere `TMetaUtils` functions are called. After discussion with @pcanal, it would seem better to place the locks in `TMetaUtils` directly rather than here. Unfortunately, when I tried doing that I discovered that I can't place dependency from `TInterpreter.h` in `TClingUtils.cxx`, so I'm not sure how we can place locks there.